apis/components/v1alpha1/kserve_types.go

```diff
@@ -77,6 +77,7 @@ type KserveCommonStatus struct {
 	// DefaultDeploymentMode is the value of the defaultDeploymentMode field
 	// as read from the "deploy" JSON in the inferenceservice-config ConfigMap
 	DefaultDeploymentMode string `json:"defaultDeploymentMode,omitempty"`
+	ServerlessAvailable   bool   `json:"serverlessAvailable"`
```

Comment:
If possible, better to use an enum rather than a boolean

Reply:
would you have a suggestion perhaps? I failed to think of a good one as to me it seems like a binary scenario

Reply:
I don't know how it is expected to be consumed, but the values this is determined from can be Managed, Unmanaged, Removed. I don't know if you need to map those options.

Reply:
Changed it to be an enum

Reply:
yeah, we moved away from bool to enum when v2 get GA because of the concern that in the future if any other options need to be considered then we will not need to change type